test/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDaoTest.java

```diff
@@ -417,65 +417,55 @@ public void getSurveyMostRecentlyPublishedVersion() {
     
     @Test(expected = EntityNotFoundException.class)
     public void getSurveyMostRecentlyPublishedVersionThrowsException() {
-        Survey firstVersion = createSurvey(new SimpleSurvey());
+        Survey firstVersion = createSurvey(new SimpleSurvey("First Survey"));
         surveyDao.getSurveyMostRecentlyPublishedVersion(studyIdentifier, firstVersion.getGuid());
     }
     
     @Test
     public void getAllSurveysMostRecentlyPublishedVersion() {
         // Get all surveys (complete set of the GUIDS, most recently published (if never published, GUID isn't included)
-        Survey firstVersion = createSurvey(new SimpleSurvey());
-        versionSurvey(firstVersion);
-        versionSurvey(firstVersion);
-        publishSurvey(studyIdentifier, firstVersion);
-
-        Survey nextVersion = createSurvey(new SimpleSurvey());
-        versionSurvey(nextVersion);
-        nextVersion = versionSurvey(nextVersion);
-        publishSurvey(studyIdentifier, nextVersion);
+        Survey firstSurvey = createSurvey(new SimpleSurvey("First Survey"));
+        firstSurvey = versionSurvey(firstSurvey);
+        firstSurvey = versionSurvey(firstSurvey);
+        firstSurvey = publishSurvey(studyIdentifier, firstSurvey);
+
+        Survey secondSurvey = createSurvey(new SimpleSurvey("Second Survey"));
+        secondSurvey = versionSurvey(secondSurvey);
+        secondSurvey = versionSurvey(secondSurvey);
+        secondSurvey = publishSurvey(studyIdentifier, secondSurvey);
+        
+        // This version is later, and should be returned
+        secondSurvey = versionSurvey(secondSurvey);
+        secondSurvey = publishSurvey(studyIdentifier, secondSurvey);
 
         // This should include firstVersion and nextVersion.
         List<Survey> surveys = surveyDao.getAllSurveysMostRecentlyPublishedVersion(studyIdentifier);
-        boolean foundFirstVersion = false;
-        boolean foundNextVersion = false;
-        for (Survey oneSurvey : surveys) {
-            if (oneSurvey.keysEqual(firstVersion)) {
-                foundFirstVersion = true;
-            } else if (oneSurvey.keysEqual(nextVersion)) {
-                foundNextVersion = true;
-            }
-        }
-        assertTrue(foundFirstVersion);
-        assertTrue(foundNextVersion);
+
+        assertEquals(2, surveys.size());
+        assertTrue(secondSurvey.keysEqual(surveys.get(0)));
+        assertTrue(firstSurvey.keysEqual(surveys.get(1)));
```

Comment:
Should we really be relying on a specific hash ordering? The original version of the test is wordier, but at least it isn't impacted by changes to hash ordering.